packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -501,6 +479,52 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
     return isEqualValues;
   }
 
+  #getCorrectNetworkClientIdByChainId(
```

Comment:
The purpose of this function is to convert an array of chainId values into an array of objects, each containing { chainId, networkClientId } pairs. If a network is selected, it assigns selectedNetworkClientId as the networkClientId; otherwise, it uses the first available network client ID. This function will be used within detectTokens to facilitate token detection in a loop